planner/start_server.py

```diff
@@ -15,44 +17,64 @@
 _ONE_DAY_IN_SECONDS = 60 * 60 * 24
 
 
+def routeresults_to_json(list: List[RouteResult]) -> str:
+    """
+    Convert a list of RouteResults to a JSON message ready to send out.
+    :param list:
+    :return:
+    """
+    routeobjs = []
+    for r in list:
+        routeobjs.append({
+            'json': json.loads(r.route),
+            'score': r.score,
+            'length': r.length
+        })
+    return json.dumps({
+        'routes': routeobjs
+    })
+
+
 class RoutePlanner(planner_pb2_grpc.RoutePlannerServicer):
 
     def PlanRoute(self, request, context):
-        data = {}
-        constraints = json.loads(request.jsonData)
-
-        print('Received PlanRoute() call. Constraints:')
-        print(constraints)
-
-        # Lat/lng of origin
-        origin = constraints.get('origin')
-        destination = constraints.get('destination')
-        desired_length = constraints.get('desiredLength')
-        if origin is not None and destination is not None:
-            orig_lat = float(origin.get('latitude'))
-            orig_lng = float(origin.get('longitude'))
-            dest_lat = float(destination.get('latitude'))
-            dest_lng = float(destination.get('longitude'))
+        req = json.loads(request.jsonData)
+        req = req['jsonData']
+        print('Received PlanRoute() call. Data:')
+        print(req)
+
+        try:
+            # Convert origins and dests to lists of tuples
+            origins = [(o['latitude'], o['longitude'])
+                       for o in req['origins']]
+            dests = [(d['latitude'], d['longitude'])
+                     for d in req['dests']]
+
+            # Neither can be empty
+            if origins == [] or dests == []:
+                raise ValueError('Origins or dests cannot be empty')
 
             with connPool.getconn() as conn:
-                # If exception is raised in this block, rollback transaction, else commit.
-                if desired_length:
-                    desired_length = float(desired_length)
-                    router = OrienteeringRouter(conn, config['gmapsApiKey'])
-                else:
+                if 'desired_dist' not in req:
                     router = Point2PointRouter(conn)
+                    routes = router.make_route(origins, dests)
 
-                linestring, length = router.make_route((orig_lat, orig_lng), (dest_lat, dest_lng), desired_length)
-                route_geometry = json.loads(linestring)
+                else:
+                    length = req['desired_dist']
+                    poi_prefs = req['poi_prefs']
+                    edge_prefs = req['edge_prefs']
+                    noptions = req['noptions']
+                    router = OrienteeringRouter(conn)
+                    routes = router.make_route(
```

Comment:
We should have a base class that takes origin, destination and kwargs of options.

Reply:
Ok, but if Point2Pointrouter and OrienteeringRouter inherit from it, Pycharm still tells me that the function signatures don't match.

Reply:
Pycharm acts wonky sometimes. It's okay to ignore that inspection warning.

Reply:
I feel that the logic of parsing options should be in the route planners, that way it's easier to have other routers.

Reply:
But I think you need to look at the options to decide which route planner to use.